components/brave_wallet/browser/blockchain_list_parser.cc

```diff
@@ -266,4 +336,243 @@ bool ParseChainList(const std::string& json, ChainList* result) {
+  //         gamers.</strong></p>\n<p>- <strong>Social media
+  //         tools</strong></p>\n<p>Messaging, news feed, referral programs.
+  //         Great networking capabilities for finding friends and
+  //         clients.</p>\n<p>- <strong>Game database</strong> For each game,
+  //         there is a detailed description along with screenshots, gameplay
+  //         videos, community reviews.</p>\n<p>- <strong>Multiple blockchain
+  //         support</strong></p>\n<p>Games and tokens based on dozens of
+  //         different blockchains.</p>\n<p>- <strong>Custom game
+  //         coins</strong></p>\n<p>Exchange in-game currencies and native
+  //         blockchain coins on the built-in crypto exchange.</p>\n<p>-
+  //         <strong>Swap and Rent</strong></p>\n<p>New possibilities for the
+  //         in-game NFT economies: swap and rent.</p>\n<p>-
+  //         <strong>Community</strong></p>\n<p>User reputation, game reviews,
+  //         item comments, user-generated game guides .</p>", "logo":
+  //         "https://dashboard-assets.dappradar.com/document/20419/gametrademarket-dapp-marketplaces-matic-logo_e3e698e60ebd9bfe8ed1421bb41b890d.png",
+  //         "link":
+  //         "https://dappradar.com/solana/marketplaces/gametrade-market-2",
+  //         "website": "https://gametrade.market/",
+  //         "chains": [
+  //           "polygon",
+  //           "solana",
+  //           "binance-smart-chain"
+  //         ],
+  //         "categories": [
+  //           "marketplaces"
+  //         ],
+  //         "socialLinks": [
+  //           {
+  //             "title": "discord",
+  //             "url": "https://discord.gg/gametrade",
+  //             "type": "discord"
+  //           },
+  //           {
+  //             "title": "medium",
+  //             "url": "https://gametrademarket.medium.com/",
+  //             "type": "medium"
+  //           },
+  //           {
+  //             "title": "twitter",
+  //             "url": "https://twitter.com/GameTradeMarket",
+  //             "type": "twitter"
+  //           },
+  //           {
+  //             "title": "youtube",
+  //             "url":
+  //             "https://www.youtube.com/channel/UCAoMHO4zQaiT-vxWOVk8IjA/videos",
+  //             "type": "youtube"
+  //           }
+  //         ],
+  //         "metrics": {
+  //           "transactions": 401926,
+  //           "uaw": 354495,
+  //           "volume": 8949.83,
+  //           "balance": 3.81
+  //         }
+  //       }
+  //     ]
+  //   }
+  // }
+
+  absl::optional<base::Value> records_v =
+      base::JSONReader::Read(json, base::JSON_PARSE_CHROMIUM_EXTENSIONS |
+                                       base::JSONParserOptions::JSON_PARSE_RFC);
+  if (!records_v || !records_v->is_dict()) {
+    VLOG(1) << "Invalid response, could not parse JSON, JSON is: " << json;
+    return false;
+  }
+
+  const auto& response_dict = records_v->GetDict();
```

Comment:
Please use generated code from `//tools/json_schema_compiler` to parse the result.
See 
```
components/brave_wallet/browser/json_rpc_responses.idl
components/brave_wallet/browser/swap_responses.idl
components/brave_wallet/common/json_rpc_requests.idl
components/brave_wallet/api/asset_ratio.idl
```
for reference